filewriter_v2/README.md

```diff
@@ -0,0 +1,388 @@
+| [fast_pixel_direction@offset](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-fast-pixel-direction-offset-attribute) | NX_NUMBER[3] | A fixed offset applied before the transformation. |
+| [fast_pixel_direction@transformation_type](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-fast-pixel-direction-transformation-type-attribute) | NX_CHAR | `"translation"` |
+| fast_pixel_direction@units | NX_LENGTH | `"m"` |
+| [fast_pixel_direction@vector](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-fast-pixel-direction-vector-attribute) | NX_NUMBER[3] | Unit vector describing the fastest varying pixel direction. |
+| [slow_pixel_direction](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-slow-pixel-direction-field) | NX_NUMBER | Size of a pixel along the slowest varying pixel direction. |
+| [slow_pixel_direction@depends_on](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-slow-pixel-direction-depends-on-attribute) | NX_CHAR | `"/entry/instrument/detector/transformations/translation"` |
+| [slow_pixel_direction@offset](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-slow-pixel-direction-offset-attribute) | NX_NUMBER[3] | A fixed offset applied before the transformation. |
+| [slow_pixel_direction@transformation_type](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-slow-pixel-direction-transformation-type-attribute) | NX_CHAR | `"translation"` |
+| slow_pixel_direction@units | NX_LENGTH | `"m"` |
+| [slow_pixel_direction@vector](https://manual.nexusformat.org/classes/applications/NXmx.html#nxmx-entry-instrument-detector-detector-module-slow-pixel-direction-vector-attribute) | NX_NUMBER[3] | Unit vector describing the slowest varying pixel direction. |
+
+## /entry/instrument/detector/transformations
+
+| Name | NeXus Type | Description |
+|---|---|---|
+|  |  |
+| @NX_class | NX_CHAR | [`"NXtransformations"`](https://manual.nexusformat.org/classes/base_classes/NXtransformations.html) |
+| translation | NX_FLOAT | Distance from the sample to the detector. |
+| translation@depends_on | NX_CHAR | `"."` |
+| translation@offset | NX_NUMBER[3] | A fixed offset applied before the transformation. |
+| translation@transformation_type | NX_CHAR | `"translation"` |
+| translation@units | NX_LENGTH | `"m"` |
+| translation@vector | NX_NUMBER[3] | Unit vector describing the direction of the translation from the sample to the detector. |
+| two_theta | NX_FLOAT[nP] | Starting $2\theta$ value for each frame. |
+| two_theta@depends_on | NX_CHAR | `"/entry/sample/transformations/translation"` |
```

Comment:
I think this is a bug. This should not point to the sample transformations group, but to the translation here in the detector transformations group. I will open a PR for wildhaus and when that is accepted, change these docs accordingly.

Reply:
Do we know what `two_theta` means exactly? Is it documented somewhere? Otherwise, it's kind of hard for me to know what the best action should be. Maybe all these goniometer angles should be simply "user defined" and the `depends_on` attribute should be avoided altogether?

Reply:
What I found is this:

![image](https://user-images.githubusercontent.com/73243774/205010466-d6b08774-bdb4-457f-8597-14b8f7ab68e4.png)

Since "our" two_theta is in the detector-transformations group, I assume we are talking about the $\alpha$ ($2\theta_{D}$), this would also match that we group it together with the distance to the sample.

I am not 100% sure though.

Reply:
screenshot is taken from this [paper](http://prism.mit.edu/xray/documents/Bruker%20GADDS%20Intro%20to%202D%20Diffraction.pdf)

Reply:
@diego-gaemperle also found this:
![two_theta](https://user-images.githubusercontent.com/73243774/205029645-1c729c73-239c-423d-9d30-62759eaa9fac.png)

After that, I actually think detector/transformations/translation needs to depend_on two_theta, since in this case (opposed to the detector/geometry case) we want $T_{R} \cdot T_{t}$, with notation as in [NXtransformations](https://manual.nexusformat.org/classes/base_classes/NXtransformations.html#nxtransformations).

Reply:
The second diagram implies that translation would have to `depend_on` `two_theta`, as you noted. However, the first diagram does not seem consistent with our defined lab coordinate system. Particularly, in the first diagram the sample starts at X_L and is rotated around Z_L. In our system, the detector face has a default normal of -Z_L. We would require Y_L to be the axis of rotation for two*theta?

Reply:
As to the original bug: We don't write the dataset `"/entry/sample/transformations/translation"` right? If not, then obviously `two_theta` should not depend on it. We should remove the `two_theta@depends_on` attribute.

Reply:
We came to the same conclusion. We removed the `two_theta@depends_on` for now, but we will ask people who are actually using this what they would expect and what their interpretation is.